tests/testthat/test-shinytest2-data_summary.R

```diff
@@ -0,0 +1,108 @@
+  skip_if_too_deep(5)
+  app <- TealAppDriver$new(
+    data = teal.data::teal_data(x = 1),
+    modules = example_module()
+  )
+
+  testthat::expect_identical(
+    as.data.frame(app$get_active_data_summary_table()),
+    data.frame(
+      `Data Name` = c('x'),
+      check.names = FALSE
+    )
+  )
+
+  app$stop()
+})
+
+
+testthat::test_that("e2e: data summary is displayed with 2 columns data without keys", {
+  skip_if_too_deep(5)
+  app <- TealAppDriver$new(
+    data = simple_teal_data(), # iris, mtcars
+    modules = example_module()
+  )
+
+  testthat::expect_identical(
+    as.data.frame(app$get_active_data_summary_table()),
+    data.frame(
+      `Data Name` = c('iris', 'mtcars'),
+      Obs = c('150/150', '32/32'),
+      check.names = FALSE
+    )
+  )
+
+  app$stop()
+})
+
+testthat::test_that("e2e: data summary is displayed with 3 columns for data with join keys", {
+  skip_if_too_deep(5)
+
+  data <- teal.data::teal_data(mtcars1 = mtcars, mtcars2 = data.frame(am = c(0,1), test = c('a', 'b')))
+
+  teal.data::join_keys(data) <- teal.data::join_keys(
+    teal.data::join_key('mtcars2', 'mtcars1', keys = c('am'))
+  )
+
+  app <- TealAppDriver$new(
+    data = data,
+    modules = example_module()
+  )
+
+  testthat::expect_identical(
+    as.data.frame(app$get_active_data_summary_table()),
+    data.frame(
+      `Data Name` = c('mtcars1', 'mtcars2'),
+      Obs = c('32/32', '2/2'),
+      Subjects = c('2/2', '2/2'),
+      check.names = FALSE
+    )
+  )
+
+  app$stop()
+})
+
+testthat::test_that(
+  "e2e: data summary is displayed properly if teal_data include data.frames with join keys, MAE objects and vectors", {
+    testthat::skip_if_not_installed("MultiAssayExperiment")
+    skip_if_too_deep(5)
+
+    data <- teal.data::teal_data() |>
+      within(
+        {
+          mtcars1 <- mtcars
+          mtcars2 = data.frame(am = c(0,1), test = c('a', 'b'))
+          iris <- iris
+          library(MultiAssayExperiment)
+          data("miniACC", package = "MultiAssayExperiment", envir = environment())
+          CO2 <- CO2
+          factors <- names(Filter(isTRUE, vapply(CO2, is.factor, logical(1L))))
+          CO2[factors] <- lapply(CO2[factors], as.character)
+        }
+      )
+
+    teal.data::join_keys(data) <- teal.data::join_keys(
+      teal.data::join_key('mtcars2', 'mtcars1', keys = c('am'))
+    )
+
+    app <- TealAppDriver$new(
+      data = data,
+      modules = example_module()
+    )
+
+    testthat::expect_identical(
+      as.data.frame(app$get_active_data_summary_table()),
+      data.frame(
+        `Data Name` = c(
+          "CO2", "iris", "miniACC", "- RNASeq2GeneNorm", "- gistict", "- RPPAArray", "- Mutations", "- miRNASeqGene",
+          "mtcars1", "mtcars2", "factors"
+        ),
+        Obs = c("84/84", "150/150", "", "198/198", "198/198", "33/33", "97/97", "471/471", "32/32", "2/2", ""),
+        Subjects = c("", "", "92/92", "79/79", "90/90", "46/46", "90/90", "80/80", "2/2", "2/2", ""),
```

Comment:
this is hedious. thinking about a better way to test this without the need to `dput(app$get_active_data_summary_table())`